runtime/metricsview/astsql.go

```diff
@@ -102,42 +118,51 @@ func (b *sqlBuilder) writeSelect(n *SelectNode) error {
 	if n.FromTable != nil {
 		b.out.WriteString(*n.FromTable)
 
-		// Add unnest joins. We only and always apply these against FromPlain (ensuring they are already unnested when referenced in outer SELECTs).
+		// Add unnest joins. We only and always apply these against FromTable (ensuring they are already unnested when referenced in outer SELECTs).
 		for _, u := range n.Unnests {
 			b.out.WriteString(", ")
 			b.out.WriteString(u)
 		}
 	} else if n.FromSelect != nil {
-		b.out.WriteByte('(')
-		err := b.writeSelect(n.FromSelect)
-		if err != nil {
-			return err
-		}
-		b.out.WriteString(") ")
-		b.out.WriteString(n.FromSelect.Alias)
-
-		for _, ljs := range n.LeftJoinSelects {
-			err := b.writeJoin("LEFT", n.FromSelect, ljs)
-			if err != nil {
-				return err
+		if n.FromSelect.IsCTE {
+			b.out.WriteString(n.FromSelect.Alias)
+			if n.JoinComparisonSelect != nil {
+				err := b.writeJoin(n.JoinComparisonType, n.FromSelect, n.JoinComparisonSelect)
+				if err != nil {
+					return err
+				}
 			}
-		}
-
-		if n.SpineSelect != nil {
-			err := b.writeJoin("RIGHT", n.FromSelect, n.SpineSelect)
+		} else {
```

Comment:
Shouldn't the `if` statement only cover the `FromSelect`? As it is right now, it duplicates the join select handling in both the `if` and `else` case, and it doesn't cover the left and spine joins in the `if` case.